descwl_coadd/tests/test_coadd.py

```diff
@@ -163,7 +170,122 @@ def test_coadds_with_interpolator_smoke(buff):
+    bdata = sim_data['band_data']
+    for band in bands:
+        assert band in bdata
+        exps = bdata[band]
+
+        # Check that there are pixels to be interpolated.
+        bit_mask = exps[0].mask.getPlaneBitMask(our_interpolator.bad_mask_planes)
+        assert any((exp.mask.array & bit_mask).any() for exp in exps)
+
+        # Since interpolation changes the exposures in place, make a deep
+        # copy before making the coadds.
+        coadd_data = make_coadd(
+            exps=[exp.clone() for exp in exps],
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=our_interpolator,
+        )
+
+        dm_coadd_data = make_coadd(
+            exps=exps,
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=dm_interpolator,
+        )
+
+        # Compare only the main exposure since the noise realizations are
+        # different and mfrac is not affected by the interpolator.
+        if flip_xy:
+            # Exact agreement is expected only with flipXY config set to True.
+            np.testing.assert_array_equal(
+                dm_coadd_data['coadd_exp'].image.array,
+                coadd_data['coadd_exp'].image.array,
+            )
+        else:
+            with np.testing.assert_raises(AssertionError):
+                np.testing.assert_array_equal(
+                    dm_coadd_data['coadd_exp'].image.array,
+                    coadd_data['coadd_exp'].image.array,
+                )
+            # The agreement is not exact due to the different triangulation.
+            # The test is intended to benchmark the disagreement due to a
+            # different, but equally valid, choice of coordinates.
+            # | actual - desired | < atol + rtol * |desired|
+            with subtests.test(msg='Benchmark differences from DM interpolator'):
+                np.testing.assert_allclose(
+                    actual=dm_coadd_data['coadd_exp'].image.array,
+                    desired=coadd_data['coadd_exp'].image.array,
+                    atol=0.41,
+                    rtol=0.0,
+                )
+                np.testing.assert_allclose(
+                    actual=dm_coadd_data['coadd_exp'].image.array,
+                    desired=coadd_data['coadd_exp'].image.array,
+                    atol=0.41,
+                    rtol=0.0,
+                )
+                np.testing.assert_allclose(
+                    actual=dm_coadd_data['coadd_exp'].image.array,
+                    desired=coadd_data['coadd_exp'].image.array,
+                    atol=0.0,
+                    rtol=1653.16,
```

Comment:
Why do we allow such a large rtol?

Reply:
All of these values are empirically set based on observed differences and are intended to act as benchmarking rather than tests. For small pixel values close to zero, this shouldn't be surprising but this is just how sensitive CTInterpolator is to the choice of coordinate frames.

Reply:
this test is better specified in atol anyways - the values cross zero